docker/Dockerfile

```diff
@@ -4,6 +4,7 @@ ARG BASE_IMAGE
 FROM $BASE_IMAGE AS base
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ARG ROS_DISTRO
+ARG SETUP_ARGS
```

Comment:
Should we move SETUP_ARGS to base image as well?
I think the argument is made for the users to add custom options to run ./setup-dev-env.sh which shouldn't affect the base image.

Reply:
This `SETUP_ARGS` determines whether to install the CUDA drivers, so it is necessary here. 
However, the actual name given, `--no-nvidia`, is misleading, so I thought it should be renamed.
https://github.com/autowarefoundation/autoware/blob/main/.github/workflows/health-check.yaml#L41
https://github.com/autowarefoundation/autoware/blob/main/.github/workflows/docker-build-and-push.yaml#L28

Reply:
I see. I misunderstood the value there. Thanks for the explanation.